pkg/collector/tcp.go

```diff
@@ -85,21 +85,18 @@ func (cp *CollectingProcess) handleTCPClient(conn net.Conn) {
 				length, err := getMessageLength(bytes.NewBuffer(buffBytes))
 				if err != nil {
 					klog.Error(err)
-					cp.deleteClient(address)
-					return
+					break
 				}
 				if size < length {
 					klog.Errorf("Message length %v is larger than size read from buffer %v", length, size)
-					cp.deleteClient(address)
-					return
+					break
 				}
 				size = size - length
 				// get the message here
 				message, err := cp.decodePacket(bytes.NewBuffer(buffBytes[0:length]), address)
 				if err != nil {
 					klog.Error(err)
-					cp.deleteClient(address)
-					return
+					break
```

Comment:
same here

Reply:
Do you mean `buffBytes` or `buff`? In my understanding reading on connections should be an overwrite to the buffer and size can determine how long we need to read and decode.

Reply:
I meant buffBytes. I know that you are reinitializing, but It is better to make it nil explicitly as we are allocating the buffer again and again.

Reply:
Make sense. Updated.